alfasim_sdk/types.py

```diff
@@ -132,16 +149,29 @@ class MultipleReference(BaseField):
     :ivar alfasim_sdk.types.ALFAsimType container_type:
         Property that indicates which type of data the MultipleReference should hold.
     """
+    ref_type = attrib()
+    container_type = attrib(default=None, validator=[optional(instance_of(str)), optional(check_string_is_not_empty)])
 
-    container_type = attrib()
+    def __attrs_post_init__(self):
+        if issubclass(self.ref_type, ALFAsimType):
+            self.container_type = attr.fields(self.ref_type).container_type.default
+        else:
+            if self.container_type is None:
+                raise TypeError(
+                    f"The container_type field must be filled when ref_type is a class decorated with 'container_model'")
 
-    @container_type.validator
-    def check(self, attr: Attribute, value: Type[ALFAsimType]) -> None:
-        if not isinstance(value, type):
-            raise TypeError(f"{attr.name} must be a class, got {type(value).__name__}")
+    @ref_type.validator
+    def check(self, attr: Attribute, value) -> None:
+        if type(value) is not type:
+            raise TypeError(f"{attr.name} must be a class")
 
         if not issubclass(value, ALFAsimType):
-            raise TypeError(f"{attr.name} must be a valid ALFAsimType, got {type(value).__name__}")
+            if not hasattr(value, "_alfasim_metadata"):
+                raise TypeError(f"{attr.name} must be an ALFAsim type or a class decorated with 'data_model'")
+
+            if value._alfasim_metadata["model"] is not None:
```

Comment:
Hmm having a value different than None in this attribute means it is a container_model decorated class? Is that supposed to be obvious and I'm missing something? If not, what do you think about adding a comment like "data model classes will have model set to None"

Reply:
Basically, the difference between `data_model` and `container_model` is the "model attribute".

I would not say that it's an "obvious" statement, but this is a behavior that we are following since the beginning. Note that there isn't any enforcement on `container_model` perhaps we should 😬 

The docstring from `data_model` and `container_model` describes try to explain this behavior.

From the `container_model`
```python
1) Application Required:
    All properties that are required from the application can be accessed from _alfasim_metadata.

    Currently, the following options are available:
        - caption: A text to be displayed
        - icon: Name of the icon available at resource folder to be used over the TreeStructure
        - model: A reference for a user-declared class that has the @data_model decorator.
```

from the `data_model`
```python
 1) Application Required:
    All properties that are required from the application can be accessed from _alfasim_metadata.

    Currently, the following options are available:
        - caption: A text to be displayed
        - icon: Name of the icon available at resource folder to be used over the TreeStructure
        - model: None (data_model cannot reference another model).
```

Reply:
Hmm ok. The docstrings are pretty clear, but I still would add some quick doc close to that code.